src/tests/ftest/util/server_utils.py

```diff
@@ -409,7 +409,8 @@ def get_detected_engine_count(self, sub_process):
         #   - the expected number of pattern matches are detected (success)
         #   - the time out is reached (failure)
         #   - the subprocess is no longer running (failure)
-        while not complete and not timed_out and sub_process.poll() is None:
+        while not complete and not timed_out \
+                and (sub_process is not None and sub_process.poll() is None):
```

Comment:
```suggestion
                and (sub_process is None or sub_process.poll() is None):
```

Reply:
Thanks for the comment, please see the updated PR.